### PR TITLE
bug fixes PR 1680

### PR DESCRIPTION
**Pull request by grumpyTofu**
#1680

### CODE REVIEW

**Comment by phryneas** [comment]
Thanks for all the testing!

As for the variable name problem:

I'd prefer to keep the behaviour as it is, but fixing the bug with the rename-on-export.

When we keep the names predictable within the file, we can't run into weird naming conflicts.

That way the import and export could for example have the same name without any problem.

So, the idea would be

```js

import { importName as api } from './importFile'

const injectedRtkApi = api.injectEndpoints({ ... })

export { injectedRtkApi as exportName }

export { useMyQuery } = injectedRtkApi

```

File: packages/rtk-query-codegen-openapi/package.json [hunk 1]
```diff
@@ -1,5 +1,5 @@
 {
-  "name": "@rtk-query/codgen-openapi",
```
Comment (phryneas) [inline packages/rtk-query-codegen-openapi/package.json:2]: Good catch! That would have been weird :rofl:

**Comment by grumpyTofu** [comment]
@phryneas Good call, I didn't even think about someone specifying them as the exact same name in their config, but now that you mention it, I could see that happening.

**Comment by phryneas** [comment]
thanks a lot!

**Pull request merged**